src/components/input-panel/compiled-spec-display/index.js

```diff
@@ -19,30 +32,52 @@ class CompiledSpecDisplay extends React.Component {
     this.setState({height});
   }
 
+  handleClick () {
+    this.setState(preState => ({
+      showCompiledSpec: !preState.showCompiledSpec
+    }));
+  }
+      
   render () {
-    return (
-        <div style={{width: '100%'}}>
-          <AceEditor
-            mode='json'
-            theme='github'
-            showGutter={true}
-            key={JSON.stringify(this.state)}
-            width={'100%'}
-            height={this.state.height + 'px'}
-            value={JSON.stringify(this.props.value)}
+    if (this.state.showCompiledSpec) {
+      let toggleStyleUp = Object.assign({}, toggleStyle, {
+        position: 'absolute',
+        bottom: this.state.height + 'px'
+      });
+      return (
+        <div style={{width: '100%', position: 'absolute', bottom: '0px'}}>
+          <div
+            style={toggleStyleUp}
+            onClick={this.handleClick.bind(this)}>Read only, click to hide spec </div>
+            <MonacoEditor 
+              mode='json'
+              theme='github'
+              showGutter={true}
+              key={JSON.stringify(this.state)}
+              width={'100%'}
+              height={this.state.height}      
+              // value={JSON.stringify(this.props.value)}
+              value={this.props.value}
```

Comment:
The properties for `MonacoEditor` are slightly different from the properties that the old `AceEditor` expected. Check out https://github.com/vega/vega-editor/blob/redux/src/components/input-panel/spec-editor/renderer.js#L61-L68 to make sure that syntax highlighting, etc. works